File: 0013_Roman2Int.cpp


class Solution {
    public:
    int romanToInt(string s) {
        unordered_map < char, int > roma = {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        }
        int sum = roma[s.back()]
        for (int i=s.length() - 2
             i >= 0
- -i){// 倒数第二个元素开始
            if (roma[s[i]] < roma[s[i + 1]])
            sum -= roma[s[i]]
            else sum += roma[s[i]]
       }
        return sum
    }
};

File: 0013_Roman2Int.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """  # learn from ibeonlyone
        return s.count('I') + s.count('V') * 5 + s.count('X') * 10 + s.count('L') * 50 + s.count('C') * 100\
            + s.count('D') * 500 + s.count('M') * 1000 - s.count('IV') * 2 - s.count('IX') * 2 - s.count('XL') * 20\
            - s.count('XC') * 20 - s.count('CD') * 200 - s.co
